api/handlers.go

```diff
@@ -16,34 +16,21 @@ import (
 	reporef "github.com/qri-io/qri/repo/ref"
 )
 
-// DatasetHandlers wraps a requests struct to interface with http.HandlerFunc
-type DatasetHandlers struct {
-	lib.DatasetMethods
-	inst     *lib.Instance
-	ReadOnly bool
-}
-
-// NewDatasetHandlers allocates a DatasetHandlers pointer
-func NewDatasetHandlers(inst *lib.Instance, readOnly bool) *DatasetHandlers {
-	h := DatasetHandlers{inst.Dataset(), inst, readOnly}
-	return &h
-}
-
 // GetHandler is a dataset single endpoint
-func (h *DatasetHandlers) GetHandler(routePrefix string) http.HandlerFunc {
+func GetHandler(inst *lib.Instance, routePrefix string) http.HandlerFunc {
```

Comment:
probably most controversial thing in this pr, removing `DatasetHandlers` and passing an `inst` directly into the `GetHandler`

Reply:
I'm totally fine with it.